benchmarks/README.md

```diff
@@ -0,0 +1,32 @@
+# Benchmarks
+
+Welcome to the benchmarks! This folder is for code benchmarking (e.g. components rather than end-to-do testing).
+The intent is to benchmark areas we think are interesting and measure improvements as well as ensuring we don't unintentionally regress over time.
+
+There's a lot of things that would be nice to have in benchmark form as we evaluate improving various parts of pipeline performance like async/await elimination,
+`Task` vs. `ValueTask`, allocations, general algorithmic improvements, etc. This is where those assessments live.
+
+To run benchmarks (from solution root - otherwise shorten the project path!):
```

Comment:
Benchmarks are still going to be comparing 2 method implementations on same build. correct? Not comparing it to the numbers for same method from previous build (I see we don't have any infra to persist that info). Is that something we plan on setting up in future?

Reply:
We have the flexibility either way, long term: yes I'd like to auto-run and detect regressions (or improvements!), but for the moment what a PR author could do is: run the benchmark against dev and then again with their changes and post the results as an intro to getting this going at all. Stage 1 is making at least that much easy, then we can iterate baking them into the process more. ...not sure I phrased that well though, does it make sense?

Reply:
Yes. makes sense. Thank you!